### DISCUSSION TITLE
3.0.0-beta4 smartcard visibility

### DISCUSSION THREAD

**Discussion by aebenjam** [FreeRDP/FreeRDP#9438] [Q&A]
I'm working from commit 9cff6e, post [3.0.0-beta4](https://github.com/FreeRDP/FreeRDP/releases/tag/3.0.0-beta4).  I built the flatpak using the [compilation instructions](https://github.com/FreeRDP/FreeRDP/wiki/Compilation) and added the tweaks from [akallabeth](https://github.com/akallabeth) so that opensc is part of the flatpak build.   I'm running on Fedora 37.

I'm trying to do Smartcard+NLA logins and, as a first step, trying to just view my smartcard/certs to make sure that part is working.  But I'm not seeing it:

```
flatpak run com.freerdp.FreeRDP /list:smartcard
X11 /app/bin/xfreerdp
[17:19:32:257] [4:00000004] [WARN][com.freerdp.smartcardlogon] - [list_provider_keys]: Microsoft Base Smart Card Crypto Provider [] no certificates found
smartcard reader detected, listing 0 certificates:
```

I get that both with and without my smartcard plugged in.  (I know the smartcard is working.  I can use it in other ways, and it's being properly passed-through using my freerdp 2.x client.  I just can't do smartcard+NLA logins.)

The smartcard is a yubikey, in case that helps.  Do I need to include a parameter pointing the software at /usr/lib64/libykcs11.so.2 or something?  (I tried some of the parameters/options, but couldn't divine the right way to do that.)

I couldn't find documentation/examples.  Apologies if I've missed something obvious.

Thanks much,

Adam

**Comment by akallabeth** [comment]
@aebenjam well...
1. check that the certificates are actually valid for smartcard logon 
2. if you need custom libraries, better not use the `flatpak` as you will have problems due to the sandbox
3. documentation is sparse as this is a complicated matter and nobody had time to write it down yet, so any pull request is welcome here ;)

**Comment by aebenjam** [comment]
I can confirm that the certs are valid for smartcard logon.  I can (using the 2.x client) present them across an RDP session and then use them (chained) to a third machine and they work fine.  I just can't use them directly for NLA + Smartcard.  (Hence my interest in the 3.x beta.)

Understood re: sandbox.  Is this something that will get solved (or I can solve now) during a flatpak build process?  If not, I can't see any Yubikey owners being able to utilize their smartcards during logon (with the flatpak) unless there's a generic library that will instead suffice.

Understood re: documentation.  I can't offer to contribute until I can get it working myself.  ;)

I can see about building a binary package instead of a flatpak...

Thanks,

Adam

**Reply by akallabeth** [reply]
@aebenjam ok, you got certificates, but are they valid for NLA logon purposes? (they are not necessarily identical)
as for yubikeys, just as documentation, this is new and needs refinement ;)

**Comment by hardening** [comment]
For now we filter on certs with RSA keys, we don't support EC. Can you check your certificates/keys regarding this ?

**Comment by hardening** [comment]
@akallabeth ok ok I definitely need to write a documentation on smartcard logon
